builtin/common/misc_helpers.lua

```diff
@@ -238,6 +238,47 @@ function math.sign(x, tolerance)
 end
 
 --------------------------------------------------------------------------------
+function math.intpow(x, y)
```

Comment:
I'd prefer if you didn't add this function and instead wrote out the values manually. Its not really implemented efficiently (doing y many multiplications), an efficient implementation would look like this: https://github.com/rust-num/num/blob/c8ed8ff87be3882854aa6f4d780bd34e3613545d/traits/src/pow.rs#L25

It has logarithmic many multiplications while your implementation has linear many.